packages/flutter/lib/src/gestures/events.dart

```diff
@@ -72,4 +72,33 @@ class PointerInputEvent extends InputEvent {
   final double tilt;
 
   ui.Point get position => new ui.Point(x, y);
+
+  String toString() => 'PointerInputEvent(x: $x, y:$y)';
+
+  String toStringFull() {
+    String result = "PointerInputEvent(";
+    result += "pointer: $pointer, ";
+    result += "kind: $kind, ";
+    result += "x: $x, ";
+    result += "y: $y, ";
+    result += "dx: $dx, ";
+    result += "dy: $dy, ";
+    result += "buttons: $buttons, ";
+    result += "down: $down, ";
+    result += "primary: $primary, ";
+    result += "obscured: $obscured, ";
+    result += "pressure: $pressure, ";
+    result += "pressureMin: $pressureMin, ";
+    result += "pressureMax: $pressureMax, ";
+    result += "distance: $distance, ";
+    result += "distanceMin: $distanceMin, ";
+    result += "distanceMax: $distanceMax, ";
+    result += "radiusMajor: $radiusMajor, ";
+    result += "radiusMinor: $radiusMinor, ";
+    result += "radiusMin: $radiusMin, ";
+    result += "radiusMax: $radiusMax, ";
+    result += "orientation: $orientation, ";
+    result += "tilt: $tilt)";
```

Comment:
You can get the compiler to do this string concatenation for you if you leave off the `;` and the `result += `.

```
+    String result = "PointerInputEvent("
+        "pointer: $pointer, "
+        "kind: $kind, "
+        "x: $x, "
+        "y: $y, "
```